src/ngraph/op/fused/gelu.cpp

```diff
@@ -74,3 +77,58 @@ void op::Gelu::pre_validate_and_infer_types()
                           input_element_type,
                           ").");
 }
+
+void op::Gelu::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+{
+    auto delta = deltas.at(0);
+
+    auto x = input(0).get_source_output();
+
+    adjoints.add_delta(x, delta * (make_shared<op::GeluBackpropFactor>(x)));
+}
+
+const string op::GeluBackpropFactor::type_name{"GeluBackpropFactor"};
+
+op::GeluBackpropFactor::GeluBackpropFactor(const Output<Node>& x)
+    : FusedOp({x})
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::GeluBackpropFactor::pre_validate_and_infer_types()
+{
+    element::Type input_element_type = get_input_element_type(0);
+
+    NODE_VALIDATION_CHECK(this,
+                          input_element_type.is_dynamic() || input_element_type == element::f32 ||
```

Comment:
You can use `input_element_type.is_real()` instead of comparing agains f32, f64, f16 and bf16. Same can be done for `Gelu` as well